tools/generator/Java.Interop.Tools.Generator.ObjectModel/Method.cs

```diff
@@ -107,6 +107,57 @@ internal string CalculateEventName (Func<string, bool> checkNameDuplicate)
 
 		public bool CanHaveStringOverload => IsReturnCharSequence || Parameters.HasCharSequence;
 
+		public Method Clone (GenBase declaringType)
+		{
+			var clone = new Method (declaringType);
+
+			// MethodBase
+			clone.Annotation = Annotation;
+			clone.ApiAvailableSince = ApiAvailableSince;
+			clone.AssemblyName = AssemblyName;
+			clone.Deprecated = Deprecated;
+			clone.DeprecatedSince = DeprecatedSince;
+			clone.IsAcw = IsAcw;
+			clone.Name = Name;
+			clone.Visibility = Visibility;
+			clone.LineNumber = LineNumber;
+			clone.LinePosition = LinePosition;
+			clone.SourceFile = SourceFile;
+			clone.JavadocInfo = JavadocInfo;
+
+			if (GenericArguments != null)
+				foreach (var ga in GenericArguments)
+					clone.GenericArguments.Add (ga.Clone ());
+
+			foreach (var p in Parameters)
+				clone.Parameters.Add (p.Clone ());
```

Comment:
While writing the [draft commit message](https://github.com/xamarin/java.interop/pull/1080#issuecomment-1430156950), my thoughts turn to `BaseClass<V>.doThing(V)`: wouldn't this still have a parameter type of `V`?  How is that "resolved" via `declaringType.TypeParameters`?

Rephrased, I would expect that the `clone` instance will have `clone.Parameters.Count==1`, and `clone.Parameters[0]` equivalent to:

```csharp
new Parameter (
        name: "value",
        type: "java.lang.Object",
        managedType: "Java.Lang.Object"
        isEnumified: false
);
```

but I don't see this happening.  *Should* this be happening?

Or is this instead handled by new lines 128-130 (above), wherein we added `this.GenericArguments` to `clone.GenericArguments`.  Though I'm not certain about that either, as shouldn't `GenericArguments` be empty as `BaseClass<V>.doThing(V)` isn't itself generic…